_maps/templates/pirate_ship.dmm

```diff
@@ -3,1633 +3,610 @@
+	name = "Unisex Restrooms"
 	},
-/turf/open/floor/plasteel/dark,
+/obj/effect/decal/cleanable/dirt,
+/turf/open/floor/plasteel/showroomfloor,
 /area/shuttle/pirate)
-"ay" = (
-/obj/machinery/light/small{
-	dir = 1
+"aw" = (
+/turf/closed/wall/mineral/plastitanium/nodiagonal,
+/area/shuttle/pirate/vault)
+"ax" = (
+/obj/effect/mapping_helpers/airlock/locked,
+/obj/machinery/door/airlock/vault{
+	id_tag = "piratevault";
+	name = "Vault"
 	},
-/obj/structure/table,
-/obj/effect/decal/cleanable/dirt,
 /obj/effect/decal/cleanable/dirt,
+/turf/open/floor/pod/dark,
+/area/shuttle/pirate/vault)
+"ay" = (
 /obj/effect/decal/cleanable/dirt,
-/obj/item/storage/toolbox/emergency,
-/obj/item/extinguisher,
-/turf/open/floor/plasteel/dark,
+/obj/machinery/iv_drip,
+/turf/open/floor/wood,
 /area/shuttle/pirate)
 "az" = (
-/obj/machinery/atmospherics/components/unary/vent_pump/on,
+/obj/structure/closet/crate/freezer/blood,
 /obj/effect/decal/cleanable/dirt,
-/turf/open/floor/plasteel/dark,
+/turf/open/floor/wood,
 /area/shuttle/pirate)
 "aA" = (
-/obj/machinery/atmospherics/pipe/simple/supply/hidden,
-/turf/open/floor/plasteel/dark,
-/area/shuttle/pirate)
-"aB" = (
-/obj/effect/decal/cleanable/dirt,
-/obj/machinery/airalarm{
-	pixel_y = 24;
-	req_access = null
+/obj/structure/closet/crate,
+/obj/item/grenade/smokebomb{
+	pixel_x = -4
 	},
-/turf/open/floor/plasteel/dark,
-/area/shuttle/pirate)
+/obj/item/grenade/smokebomb{
+	pixel_x = 2
+	},
+/obj/item/gun/ballistic/shotgun/automatic/combat,
+/obj/item/shield/riot,
+/obj/item/shield/riot,
+/obj/item/gun/energy/laser,
+/obj/item/gun/energy/laser,
+/obj/item/gun/energy/laser,
+/obj/item/gun/energy/laser,
+/obj/item/storage/box/lethalshot,
+/obj/item/restraints/handcuffs,
+/obj/item/restraints/handcuffs,
+/turf/open/floor/pod/light,
+/area/shuttle/pirate/vault)
+"aB" = (
+/turf/open/floor/pod/dark,
+/area/shuttle/pirate/vault)
 "aC" = (
-/obj/machinery/light/small{
-	dir = 1
+/obj/structure/closet/crate,
+/obj/item/storage/bag/money,
+/obj/item/stack/sheet/mineral/gold{
+	amount = 3;
+	pixel_x = -2;
+	pixel_y = 2
 	},
-/obj/structure/table,
-/obj/effect/decal/cleanable/dirt,
-/obj/item/paper_bin{
-	pixel_x = -1;
-	pixel_y = 6
+/obj/item/stack/sheet/mineral/silver{
+	amount = 8;
+	pixel_x = 2;
+	pixel_y = -1
 	},
-/obj/item/folder,
-/obj/item/pen,
-/turf/open/floor/plasteel/dark,
-/area/shuttle/pirate)
+/obj/item/coin/gold,
+/obj/item/coin/gold,
+/obj/item/coin/gold,
+/obj/item/coin/silver,
+/obj/item/coin/silver,
+/obj/item/coin/silver,
+/obj/item/coin/silver,
+/obj/item/coin/silver,
+/turf/open/floor/pod/light,
+/area/shuttle/pirate/vault)
 "aD" = (
-/obj/structure/sign/departments/medbay/alt,
-/turf/closed/wall/mineral/plastitanium/nodiagonal,
-/area/shuttle/pirate)
-"aE" = (
-/obj/effect/decal/cleanable/dirt,
-/turf/open/floor/plasteel/white/side{
-	dir = 9
+/obj/effect/mapping_helpers/airlock/locked,
+/obj/machinery/door/airlock/external{
+	id_tag = "pirateportexternal"
+	},
+/obj/docking_port/mobile/pirate{
```

Comment:
since this is a unique instance only used for this map, i'd say just move all these vars onto the object itself.

Reply:
I'll keep it in code. Code organization is much better than the half a femtosecond performance that we get from doing this.